experimental/e2e/storage/har.go

```diff
@@ -222,12 +219,13 @@ func (s *HARStorage) Entries() []*Entry {
 	return entries
 }
 
-// Delete removes the HAR entry with the given ID.
-func (s *HARStorage) Delete(id string) bool {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-	for i, e := range s.har.Log.Entries {
-		if e.Comment == id {
+// Delete removes the HAR entry matching the given Request.
+func (s *HAR) Delete(req *http.Request) bool {
+	for i, e := range s.Entries() {
+		if res := e.Match(req); res != nil {
+			defer res.Body.Close()
+			s.lock.Lock()
+			defer s.lock.Unlock()
```

Comment:
Is clearer to just call `s.lock.Unlock()` after `s.har.Log.Entries = append...` as we're returning then anyway, so no need to `defer`

Reply:
addressed in [63c3326](https://github.com/grafana/grafana-plugin-sdk-go/pull/473/commits/63c332602e8904fb391f48b6e5275fc530b98480)